flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarFunctionCallGen.scala

```diff
@@ -66,7 +69,15 @@ class ScalarFunctionCallGen(scalarFunction: ScalarFunction) extends CallGenerato
       boxedTypeTermForType(returnType)
     }
     val resultTerm = ctx.addReusableLocalVariable(resultTypeTerm, "result")
-    val evalResult = s"$functionReference.eval(${parameters.map(_.resultTerm).mkString(", ")})"
+    val evalResult =
+      if (returnType.getTypeRoot == LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE) {
```

Comment:
`Types.TIMESTAMP can be represented as long` is a very old document, now, `TIMESTAMP_WITHOUT_TIME_ZONE` can not be represented as long, you can take a look to `TimestampType.INPUT_OUTPUT_CONVERSION`.

Reply:
I have no idea whether we should disable a UDX with `long` parameter to accept a `Timestamp`. What do you think @KurtYoung ?